src/InternalEvents.tsx

```diff
@@ -47,9 +49,9 @@ function useToggleHandler() {
       }
     }
 
-    window.addEventListener("keydown", handleKeyDown);
-    return () => window.removeEventListener("keydown", handleKeyDown);
-  }, [query]);
+    window.addEventListener("keydown", handleKeyDown, true);
+    return () => window.removeEventListener("keydown", handleKeyDown, true);
```

Comment:
🍖  Setting `useCapture` to `true` here makes sure that this `keydown` event will fire before others, even when other `keydown` event listeners are registered higher up in the DOM tree. 

This ensures our `event.stopPropagation` actually stops the event from being captured by, say, a modal that is open behind kbar.